python/lsst/sims/catalogs/generation/db/CompoundCatalogDBObject.py

```diff
@@ -0,0 +1,185 @@
+                if new_row[1] is None:
+                    new_row[1] = row[0]
+                self.columns.append(tuple(new_row))
+                column_names.append(new_row[0])
+
+                # 25 August 2015
+                # This is a modification that needs to be made in order for this
+                # class to work with GalaxyTileObj.  The column galaxytileid in
+                # GalaxyTileObj is removed from the query by query_columns, but
+                # somehow injected back in by the query procedure on fatboy. This
+                # leads to confusion if you try to query something like
+                # galaxyAgn_galaxytileid.  We deal with that by removing all column
+                # names like 'galaxytileid' in query_columns, but leaving 'galaxytileid'
+                # un-mangled in self.columns so that self.typeMap knows how to deal
+                # with it when it comes back.
+                if row[0] not in column_names and (row[1] is None or row[1]==row[0]):
+                    self.columns.append(row)
+                    column_names.append(row[0])
+
+
+    def _make_dbTypeMap(self):
+        """
+        Construct the self.dbTypeMap member by concatenating the self.dbTypeMaps
+        from the input CatalogDBObjects.
+        """
+        self.dbTypeMap = {}
+        for dbo in self._dbObjectList:
+            for col in dbo.dbTypeMap:
+                if col not in self.dbTypeMap:
+                    self.dbTypeMap[col] = dbo.dbTypeMap[col]
+
+
+    def _make_dbDefaultValues(self):
+        """
+        Construct the self.dbDefaultValues member by concatenating the
+        self.dbDefaultValues from the input CatalogDBObjects.
+        """
+        self.dbDefaultValues = {}
+        for dbo, dbName in zip(self._dbObjectList, self._nameList):
+            for col in dbo.dbDefaultValues:
+                self.dbDefaultValues['%s_%s' % (dbName, col)] = dbo.dbDefaultValues[col]
+
+
+    def _validate_input(self):
+        """
+        Verify that the CatalogDBObjects passed to the constructor
+        do, indeed, query the same table of the same database.
+
+        Also verify that this class is designed to query the tables
+        it is being used on (in cases where a custom query_columns
+        has been implemented).
+        """
+        hostList = []
+        databaseList = []
+        portList = []
+        driverList = []
+        tableList = []
+        objidList = []
+        for dbo in self._dbObjectList:
+            if dbo.host not in hostList:
+                hostList.append(dbo.host)
+            if dbo.database not in databaseList:
+                databaseList.append(dbo.database)
+            if dbo.port not in portList:
+                portList.append(dbo.port)
+            if dbo.driver not in driverList:
+                driverList.append(dbo.driver)
+            if dbo.tableid not in tableList:
+                tableList.append(dbo.tableid)
+
```

Comment:
actually... I'm going to do the exact opposite and add a blank line between each independent if statement to improve readability (otherwise, it looks like this is a series of if-else-else-else... statements)